src/componentsNext/FilePicker/FilepickerItem.tsx

```diff
@@ -0,0 +1,66 @@
+import { FC, Fragment } from "react";
+import {
+  FilepickerClose,
+  FilepickerListItem,
+  FilepickerSeparator,
+} from "~/design/Filepicker";
+
+import { NodeSchemaType } from "~/api/tree/schema/node";
+import { fileTypeToIcon } from "~/api/tree/utils";
+import { twMergeClsx } from "~/util/helpers";
+
+export type FilepickerItemType = {
+  results: NodeSchemaType[];
+  selectable: ((node: NodeSchemaType) => boolean) | undefined;
+  setPath: (path: string) => void;
+  setInputValue: (value: string) => void;
+  onChange: (path: string) => void;
+};
+
+export const FilepickerItem: FC<FilepickerItemType> = ({
+  results,
+  selectable,
+  setPath,
+  setInputValue,
+  onChange,
+}) => (
+  <>
+    {results.map((file) => {
+      const isSelectable = selectable?.(file) ?? true;
+      return (
+        <Fragment key={file.name}>
+          {file.type === "directory" ? (
+            <div
+              onClick={() => {
+                setPath(file.path);
+              }}
+              className="h-auto w-full cursor-pointer text-gray-11 hover:underline focus:bg-transparent focus:ring-0 focus:ring-transparent focus:ring-offset-0 dark:text-gray-dark-11 dark:focus:bg-transparent"
+            >
+              <FilepickerListItem icon={fileTypeToIcon(file.type)}>
+                {file.name}
+              </FilepickerListItem>
+            </div>
+          ) : (
+            <FilepickerClose
+              className={twMergeClsx(
+                "h-auto w-full text-gray-11 hover:underline dark:text-gray-dark-11",
+                !isSelectable && "cursor-not-allowed opacity-70"
+              )}
+              disabled={!isSelectable}
+              onClick={() => {
+                setPath(file.parent);
+                setInputValue(file.path);
+                onChange?.(file.path);
+              }}
+            >
+              <FilepickerListItem icon={fileTypeToIcon(file.type)}>
+                {file.name}
+              </FilepickerListItem>
+            </FilepickerClose>
+          )}
+          <FilepickerSeparator />
```

Comment:
I think we should hide this `<FilepickerSeparator />` component when it is the last list item, to avoid having two lines?

![CleanShot 2024-01-10 at 16 31 54@2x](https://github.com/direktiv/direktiv-ui/assets/121789579/1a70622b-563a-482e-88c8-9e3f8ed67640)

Reply:
Oh... Yes of course!